open-craft/opencraft
# [FAL-2123] Ocim Add playbook and block structure caching configuration

Added a course block structure caching configuration, and a playbook to run the commands to enable it(specified here https://tasks.opencraft.com/browse/FAL-1918)
The think that the playbook could be run on appserver spawn, as the commands that are executed willl not affect in a bad way the server if they are runned on each appserver deploy.

**Jira ticket**: [FAL-2123](https://tasks.opencraft.com/browse/FAL-2123)
- On openedx_config I added the EDXAPP_BLOCK_STRUCTURES_SETTINGS required for the caching
- Created a new playbook enable_course_block_structure_caching, to execute the commands established on [FAL-1918](https://tasks.opencraft.com/browse/FAL-1918)
- The playbook is added to `get_playbooks` method in `openedx_appserver` model, si it is called on the provisioning

**How to test**
- Once deployed on ocim stage, we should redeploy the instances
- Check that the S3 buckets related to the redeployed instances have been populated
- Check caching is working for the Course Outline pages of the existing courses
